### PR TITLE
Update feature tests to es6.

### PR DESCRIPTION
**Pull request by johnjbarton**
Mostly this is porting to es6 and fixups.

Some files no longer used by feature tests
are still used by a few node tests.

Tested in the browser as well.

### CODE REVIEW

File: test/featureTestRunner.js [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+// limitations under the License.
+
+
+import {NodeTraceurTestRunner} from './modular/NodeTraceurTestRunner.js';
+import {BrowserTraceurTestRunner} from './modular/BrowserTraceurTestRunner.js';
+
+import {assert, assertArrayEquals} from './asserts.js';
+export * from './asserts.js';
+
+function forEachPrologLine(s, f) {
```
Comment (arv) [inline test/featureTestRunner.js]: This looks like the code was moved from somewhere but I don't see where it got removed in this PR. Are we duplicating this code now?

File: test/featureTestRunner.js [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+Reflect.global.assertArrayEquals = assertArrayEquals;
+Reflect.global.assertHasOwnProperty = assertHasOwnProperty;
+Reflect.global.assertLacksOwnProperty = assertLacksOwnProperty;
+Reflect.global.assertNoOwnProperties = assertNoOwnProperties;
+Reflect.global.fail = fail;
+
+function importFeatureFiles(files) {
+  return new Promise((resolve) => {
+    // Bucket tests.
+    let tree = {};
```
Comment (arv) [inline test/featureTestRunner.js]: Object.create(null) or Map/StringMap?

File: test/featureTestRunner.js [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+    files.forEach(function(path) {
```
Comment (arv) [inline test/featureTestRunner.js]: Use arrow functions

File: test/featureTestRunner.js [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+      let parts = path.split('/');
```
Comment (arv) [inline test/featureTestRunner.js]: Windows?

File: test/featureTestRunner.js [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+      parts = parts.slice(3);
+      path = parts.join('/');
+      let suiteName = parts.slice(0, -1).join(' ');
+      let testName = parts[parts.length - 1];
+      if (!tree[suiteName])
+        tree[suiteName] = [];
+      tree[suiteName].push({name: testName, path: path});
```
Comment (arv) [inline test/featureTestRunner.js]: rename testName to name and you can do `{name, path}`

File: test/featureTestRunner.js [hunk 6]
```diff
@@ -0,0 +1,347 @@
...
+    });
+
+    suite('Feature Tests', function() {
+      for (let suiteName in tree) {
+        suite(suiteName, function() {
```
Comment (arv) [inline test/featureTestRunner.js]: arrows, arrows , arrows

File: test/featureTestRunner.js [hunk 7]
```diff
@@ -0,0 +1,347 @@
...
+          tree[suiteName].forEach(function(tuple) {
+            featureTest(tuple.name, './test/feature/' + tuple.path);
+          });
+        });
+      }
+    });
+
+    suite('Clone Tree Tests', function() {
+      for (let suiteName in tree) {
```
Comment (arv) [inline test/featureTestRunner.js]: A Map and for-of loop would be cleaner.

File: test/modular/TraceurTestRunner.js [hunk 8]
```diff
@@ -43,17 +43,10 @@ export class TraceurTestRunner extends Mocha6 {
...
-            numberOfFailures++;
-          });
-          runner.on('end', () => {
-            resolve(numberOfFailures);
-          });
-          runner.on('error', (ex) => {
-            reject(ex);
-          });
+      // Convert from callback to promise
+      return new Promise((resolve, reject) => {
```
Comment (arv) [inline test/modular/TraceurTestRunner.js]: This looks strange. Doesn't `run` return a promise any more?

File: test/runFeatureTests.js [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* @fileoverview Configure mocha and run the test list */
+
+import {featureTestRunner} from './featureTestRunner.js';
+
+featureTestRunner.run().catch(function(ex) {
+	console.error('featureTestRunner FAILED ' + ex, ex.stack);
+})
```
Comment (arv) [inline test/runFeatureTests.js]: newline at end of file

File: test/featureTestRunner.js (hunk 1 above)
Comment (johnjbarton) [inline test/featureTestRunner.js]: Yes, test-util.js is still used by a few misc. tests. I'll convert them next.  Here I was mostly focused on getting mocha to work with es6 modules incl. in the web version.

File: test/featureTestRunner.js (hunk 3 above)
Comment (johnjbarton) [inline test/featureTestRunner.js]: done. 16 times ;-)

File: test/featureTestRunner.js (hunk 2 above)
Comment (johnjbarton) [inline test/featureTestRunner.js]: done

File: test/featureTestRunner.js (hunk 4 above)
Comment (johnjbarton) [inline test/featureTestRunner.js]: I applied the `replace()` where the files list is built.

File: test/featureTestRunner.js (hunk 5 above)
Comment (johnjbarton) [inline test/featureTestRunner.js]: done

File: test/featureTestRunner.js (hunk 6 above)
Comment (johnjbarton) [inline test/featureTestRunner.js]: done

File: test/featureTestRunner.js (hunk 7 above)
Comment (johnjbarton) [inline test/featureTestRunner.js]: done

File: test/modular/TraceurTestRunner.js (hunk 8 above)
Comment (johnjbarton) [inline test/modular/TraceurTestRunner.js]: I moved the conversion from callbacks to promises up from this class to `Mocha6.run()` where it belongs.

File: test/runFeatureTests.js (hunk 9 above)
Comment (johnjbarton) [inline test/runFeatureTests.js]: done

**Comment by johnjbarton** [comment]
PTAL  I'll see about converting the remaining tests.

**Comment by arv** [comment]
LGTM, assuming the duplicated code gets removed in a future PR.

**Pull request closed without merging**